triton/kernels/quarot/torch_mm_test.py

```diff
@@ -0,0 +1,113 @@
+import random
+from scipy.linalg import hadamard
+import torch
+import numpy as np
+
+int8_mag_max = 127
+
+# TODO: look into gradient scaling
+
+def quantize(x: torch.HalfTensor):
```

Comment:
it's usually better to avoid the type-specific constructors, I believe the pytorch team views these as legacy. Mypy won't be able to type check a tensor constructed via torch.tensor([1,2,3], dtype=torch.half) as being a `HalfTensor`, since the dtype isn't part of the type.